### PR TITLE
[FLINK-30538][SQL gateway/client] Improve error handling of stop job operation

### PR DESCRIPTION
**Pull request by link3280**
## What is the purpose of the change

Currently, the stop-job operation produces some verbose error msg and doesn't handle exceptions in stop-without-savepoint gracefully.

This PR fixes the problem.

## Brief change log

- Wrap simple cancel with try-catch.
- Wait for simple cancel Acknowledge before returning 'OK'.
- Simplify exception message for stop job operations.

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no 
  - The runtime per-record code paths (performance sensitive): no 
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no 
  - The S3 file system connector: no 

## Documentation

  - Does this pull request introduce a new feature?  no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by link3280** [comment]
@flinkbot run azure

**Comment by link3280** [comment]
@flinkbot run azure

**Comment by link3280** [comment]
Please kindly take a look @fsk119

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java [hunk 1]
```diff
@@ -487,23 +487,22 @@ public ResultFetcher callStopJobOperation(
...
-                                                e);
+                                    } else {
+                                        clusterClient.cancel(JobID.fromHexString(jobId)).get();
+                                        return Optional.empty();
                                     }
-                                } else {
-                                    clusterClient.cancel(JobID.fromHexString(jobId));
-                                    return Optional.empty();
+                                } catch (Exception e) {
+                                    throw new FlinkException(e);
```
Comment (fsk119) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java:495]: Can we directly throw SqlExecutionExcetpion here? We can also catch FlinkException in the `runClusterAction`.

**Review by fsk119 — COMMENTED** [review]
Thanks for your update.

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java (hunk 1 above)
Comment (link3280) [reply flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationExecutor.java:495]: Good point. Fixed in https://github.com/apache/flink/pull/21582.

**Comment by link3280** [comment]
CI ran into https://issues.apache.org/jira/browse/FLINK-30328. Re-run CI.

**Comment by link3280** [comment]
@flinkbot run azure

**Comment by link3280** [comment]
@fsk119 CI turned green. Please kindly take a look at your convenience.

**Comment by link3280** [comment]
ping @fsk119 . It should be a quick one :)

**Comment by link3280** [comment]
ping @fsk119

**Review by fsk119 — APPROVED** [review]
LGTM

**Pull request merged**